# Donovan Feist

# CH EN 3603

# Homework 9

# Problem 2

Given the data in problem 1 and the same desired recovery of $CO_2$,

1. Determine the minimum value of $\frac{L^{'}}{V^{'}}$ to achieve the desired separation.

2. If the solvent flow rate is twice the minimum, how many theoretical stages are required?

3. If each stage has an efficiency of 40%, how many actual trays are required?

## part 1 solution

In [1]:
#imports
import numpy as np
import matplotlib.pyplot as plt
from scipy.optimize import fsolve


#data
y7=0.10 #mole fraction of CO2 in air-CO2 mixture that is to be absorbed
normality=5.0 #mole equivalents/L; normality of absorbing solution of triethanolamine
X0=0.04 #moles of CO2 per mole of amine solution
T=298 #K (=25 C); isothermal operating temperature of the column
α=0.784 #exit liquid contains 78.4% of the CO2 in the feed gas to the absorber (desired recovery)
plates=6 #number of theoretical plates
Y=np.array([0.003,0.008,0.015,0.023,0.032,0.043,0.055,0.068,0.083,0.099,0.12])
X=np.array([0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.10,0.11])

$Y_7$ is found with the equation

$Y_7=\frac{y_7}{1-y_7}$

In [2]:
Y7=y7/(1-y7)
print('Y7=',Y7)

Y7= 0.11111111111111112


$Y_7$ is then used with the equilibrium data to interpotate a value for $X_7$

In [3]:
X7=np.interp(Y7,Y,X)
print('X7=',X7)

X7= 0.10576719576719577


$Y_1$ is found with the equation

### $Y_1=(1-\alpha) \cdot Y_7$

In [4]:
Y1=(1-α)*Y7
print('Y1=',Y1)

Y1= 0.023999999999999997


Now that $X_7$, $Y_7$, $X_0$, and $Y_1$ are known, that minimum value for $\frac{L^{'}}{V^{'}}$ can be calculated.

In [5]:
LpminoVp=(Y7-Y1)/(X7-X0)
print(LpminoVp)

1.3245374094931617


$\boxed{\frac{L_{min}^{'}}{V^{'}}=1.32}$

## part 2 solution

The value for $\frac{L^{'}}{V^{'}}$ in this part is two times the minimum value.

In [6]:
LpoVp=2*LpminoVp
print(LpoVp)

2.6490748189863234


Now that the slope is known, $X_1$ can be found through interpolation with $Y_1$ and the equilibrium data. Once $X_1$ is known, the equation for the operating line,

### $Y_2=(X_1-X_0)\cdot \frac{L^{'}}{V^{'}} +Y_1$

can be used to find the value for $Y_2$.

The iterative process of interpolating, then using the operating line to find the next value of Y, is repeated until the value of $Y_{N+1}$ is greater than or equal to the desired value of $Y_{N+1}$. When this condition is met, the number of theoretical stages required is equal to N.

In [7]:
print('Y_desired=',Y7)
print('X0=',X0)
print('Y1=',Y1)
X1=np.interp(Y1,Y,X)
print('X1=',X1)
Y2=(X1-X0)*LpoVp+Y1
print('Y2=',Y2)
X2=np.interp(Y2,Y,X)
print('X2=',X2)
Y3=(X2-X0)*LpoVp+Y1
print('Y3=',Y3)
X3=np.interp(Y3,Y,X)
print('X3=',X3)
Y4=(X3-X0)*LpoVp+Y1
print('Y4=',Y4)
X4=np.interp(Y4,Y,X)
print('X4=',X4)
Y5=(X4-X0)*LpoVp+Y1
print('Y5=',Y5)

Y_desired= 0.11111111111111112
X0= 0.04
Y1= 0.023999999999999997
X1= 0.04111111111111111
Y2= 0.026943416465540358
X2= 0.044381573850600396
Y3= 0.03560711695515445
X3= 0.05327919723195859
Y4= 0.059177587003534385
X4= 0.07321352846425722
Y5= 0.11198512190434927


$Y_5$ is greater than the desired value for $Y_{N+1}$, which means that $\fbox{4 stages are required}$

## part 3 solution

The number of actual stages is given by the equation

$N_a=\frac{N_t}{\eta}$

In [9]:
η=0.40

Na=4/η
print(Na)

10.0


$\fbox{10 actual stages}$